lib/common/ipc_client.c

```diff
@@ -1551,3 +1548,28 @@ pcmk__ipc_is_authentic_process_active(const char *name, uid_t refuid,
     }
     return rc;
 }
+
+/*!
+ * \internal
+ * \brief Add local node name and ID to an IPC XML request
+ *
+ * \param[in,out] request  XML request to modify
+ * \param[in]     node     The local node's name
+ * \param[in]     nodeid   The local node's ID (can be 0)
+ */
+void
+pcmk__add_node_to_request(xmlNode *request, const char *node, int nodeid)
```

Comment:
Nice that this came in so handy.

For the name, I've been using "xe" in any function that manipulates XML elements. Also, this is not particularly IPC- or even request-specific; xml.c, nvpair.c, or a new nodes.c would be better. Maybe pcmk__xe_add_node().

Reply:
Renamed and added to nodes.c.